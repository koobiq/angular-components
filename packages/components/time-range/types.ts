import { KbqPopoverTrigger } from '@koobiq/components/popover';
import { DurationUnit } from '@koobiq/date-adapter';

export interface KbqRangeValue<T> {
    fromTime?: T;
    fromDate?: T;
    toTime?: T;
    toDate?: T;
}

export type KbqRange = {
    startDateTime?: string;
    endDateTime?: string;
};

export type KbqTimeRangeRange = KbqRange & KbqTimeRangeTypeContext;

export type KbqTimeRangeTypeContext = {
    type: KbqTimeRangeType;
};

/**
 * Default accepted “time‑range” identifiers.
 */
export type KbqTimeRangeType =
    | 'lastMinute'
    | 'last5Minutes'
    | 'last15Minutes'
    | 'last30Minutes'
    | 'lastHour'
    | 'last24Hours'
    | 'last3Days'
    | 'last7Days'
    | 'last14Days'
    | 'last30Days'
    | 'last3Months'
    | 'last12Months'
    | 'allTime'
    | 'currentQuarter'
    | 'currentYear'
    | 'range';

export type KbqTimeRangeTranslationType =
    | Extract<DurationUnit, 'seconds' | 'minutes' | 'hours' | 'days' | 'weeks' | 'months'>
    | 'other';

export interface KbqTimeRangeUnits {
    seconds?: number;
    minutes?: number;
    hours?: number;
    days?: number;
    weeks?: number;
    months?: number;
}

export type KbqTimeRangeTitleContext = KbqTimeRangeRange & KbqTimeRangeUnits;
export type KbqTimeRangeCustomizableTitleContext = Partial<KbqTimeRangeTitleContext> & {
    $implicit: Partial<KbqTimeRangeTitleContext> & { formattedDate: string; popover: KbqPopoverTrigger };
} & { formattedDate: string; popover: KbqPopoverTrigger };

export type KbqTimeRangeTranslateTypeMap = Record<KbqTimeRangeType, KbqTimeRangeTranslationType>;
export type KbqTimeRangeOptionContext = KbqTimeRangeTypeContext &
    KbqTimeRangeUnits & { translationType: KbqTimeRangeTranslationType };
