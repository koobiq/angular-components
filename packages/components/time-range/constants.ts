import { InjectionToken } from '@angular/core';

export const KBQ_DEFAULT_TIME_RANGE_TYPES = new InjectionToken('KBQ_DEFAULT_TIME_RANGE_TYPES');
