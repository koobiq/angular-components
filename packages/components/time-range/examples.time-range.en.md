This example shows how to customize the display of a selection item in a dialog

<!-- example(time-range-custom-trigger) -->
