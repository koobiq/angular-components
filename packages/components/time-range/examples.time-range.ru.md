Пример демонстрирует настройку отображения пункта выбора в диалоге

<!-- example(time-range-custom-option) -->
