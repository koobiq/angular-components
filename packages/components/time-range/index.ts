export * from './module';
export * from './time-range';
export * from './time-range-editor';
export * from './time-range-title';
