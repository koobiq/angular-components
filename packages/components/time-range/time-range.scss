.kbq-time-range {
    .kbq-time-range__trigger {
        display: block;
    }

    .kbq-time-range__buttons {
        display: flex;
        flex-direction: row;
        gap: var(--kbq-size-l);
    }
}

.kbq-time-range__editor {
    display: block;

    .kbq-time-range-editor__container {
        width: 100%;
    }

    .kbq-time-range-editor__predefined {
        column-count: 2;

        .kbq-time-range-editor__predefined-option {
            display: block;
            margin-bottom: var(--kbq-size-s);
        }
    }

    .kbq-time-range-editor__range {
        display: flex;
        flex-direction: column;
        gap: var(--kbq-size-s);

        .kbq-time-range-editor__predefined {
            column-count: 2;

            .kbq-time-range-editor__predefined-option {
                display: block;
            }
        }

        .kbq-time-range-editor__date-time {
            display: flex;
            flex-direction: row;
            justify-content: space-between;
            gap: var(--kbq-size-s);

            .kbq-time-range-editor__date-time-prefix {
                align-self: center;
                width: 57px;
            }
        }
    }
}

.kbq-time-range-title-as-form-field {
    white-space: nowrap;
    text-overflow: ellipsis;
    min-height: calc(var(--kbq-form-field-size-height) - 2 * var(--kbq-size-border-width));
}
