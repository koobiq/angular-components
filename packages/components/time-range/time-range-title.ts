import { NgTemplateOutlet } from '@angular/common';
import { Component, computed, inject, input, signal, TemplateRef } from '@angular/core';
import { takeUntilDestroyed, toObservable } from '@angular/core/rxjs-interop';
import { KbqTimeRangeLocaleConfig } from '@koobiq/components/core';
import { KbqIconModule } from '@koobiq/components/icon';
import { KbqLinkModule } from '@koobiq/components/link';
import { merge, skip } from 'rxjs';
import { KbqTimeRangeService } from './time-range.service';
import { KbqTimeRangeCustomizableTitleContext, KbqTimeRangeRange, KbqTimeRangeTitleContext } from './types';

/** @docs-private */
@Component({
    standalone: true,
    selector: 'kbq-time-range-title',
    imports: [
        KbqLinkModule,
        KbqIconModule,
        NgTemplateOutlet
    ],
    template: `
        @if (titleTemplate()) {
            <ng-container *ngTemplateOutlet="titleTemplate()!; context: titleContext() ?? null" />
        } @else {
            <a kbq-link pseudo>
                <span class="kbq-link__text">{{ formattedDate() }}</span>

                <i kbq-icon="kbq-calendar-o_16"></i>
            </a>
        }
    `
})
export class KbqTimeRangeTitle {
    readonly timeRange = input<KbqTimeRangeRange>();
    readonly titleTemplate = input<TemplateRef<any>>();
    readonly localeConfiguration = input.required<KbqTimeRangeLocaleConfig>();

    protected readonly formattedDate = signal<string | undefined>(undefined);

    private readonly timeRangeService = inject(KbqTimeRangeService);

    constructor() {
        merge(toObservable(this.context), toObservable(this.localeConfiguration).pipe(skip(1)))
            .pipe(takeUntilDestroyed())
            .subscribe(this.updateFormattedDate);
    }

    protected context = computed<KbqTimeRangeTitleContext | undefined>(() => {
        const timeRange = this.timeRange();

        if (!timeRange) return undefined;

        return {
            ...timeRange,
            ...this.timeRangeService.getTimeRangeTypeUnits(timeRange.type)
        };
    });

    protected titleContext = computed<KbqTimeRangeCustomizableTitleContext | undefined>(() => {
        const context = this.context();

        if (!context) return undefined;

        return {
            $implicit: context,
            ...context
        };
    });

    private updateFormattedDate = () => {
        const context = this.context();

        if (!context) {
            this.formattedDate.set('');

            return;
        }

        const timeRangeUnit = this.timeRangeService.getTimeRangeUnitByType(context.type);
        const localeConfig = this.localeConfiguration();

        if (timeRangeUnit === 'other') {
            switch (context.type) {
                case 'range': {
                    this.formattedDate.set(
                        localeConfig.title.for +
                            ' ' +
                            this.timeRangeService.dateFormatter.rangeLongDate(
                                this.timeRangeService.fromISO(context.startDateTime ?? ''),
                                this.timeRangeService.fromISO(context.endDateTime ?? '')
                            )
                    );
                    break;
                }
                case 'allTime': {
                    this.formattedDate.set(localeConfig.editor.allTime);
                    break;
                }
                case 'currentQuarter': {
                    this.formattedDate.set(localeConfig.editor.currentQuarter);
                    break;
                }
                case 'currentYear': {
                    this.formattedDate.set(localeConfig.editor.currentYear);
                    break;
                }
            }

            return;
        }

        if (!context.startDateTime) {
            this.formattedDate.set('');

            return;
        }

        this.formattedDate.set(
            localeConfig.title.for +
                ' ' +
                this.timeRangeService.dateFormatter.duration(
                    this.timeRangeService.fromISO(context.startDateTime),
                    this.timeRangeService.dateAdapter.today(),
                    [timeRangeUnit],
                    false,
                    localeConfig.durationTemplate
                )
        );
    };
}
