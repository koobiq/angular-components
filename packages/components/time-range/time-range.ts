import {
    booleanAttribute,
    ChangeDetectionStrategy,
    Component,
    computed,
    inject,
    InjectionToken,
    input,
    OnInit,
    output,
    Provider,
    signal,
    TemplateRef,
    ViewEncapsulation,
    WritableSignal
} from '@angular/core';
import { takeUntilDestroyed, toObservable } from '@angular/core/rxjs-interop';
import { ControlValueAccessor, FormControl, NgControl, ReactiveFormsModule } from '@angular/forms';
import { KbqButtonModule } from '@koobiq/components/button';
import {
    KBQ_LOCALE_SERVICE,
    KbqTimeRangeLocaleConfig,
    PopUpPlacements,
    PopUpSizes,
    ruRULocaleData
} from '@koobiq/components/core';
import { KbqPopoverModule, KbqPopoverTrigger } from '@koobiq/components/popover';
import { KbqTimeRangeEditor } from './time-range-editor';
import { KbqTimeRangeTitle } from './time-range-title';
import { KbqTimeRangeService } from './time-range.service';
import { KbqRangeValue, KbqTimeRangeCustomizableTitleContext, KbqTimeRangeRange, KbqTimeRangeType } from './types';

/** Localization configuration provider. */
export const KBQ_TIME_RANGE_LOCALE_CONFIGURATION = new InjectionToken<KbqTimeRangeLocaleConfig>(
    'KBQ_TIME_RANGE_LOCALE_CONFIGURATION',
    { factory: () => ruRULocaleData.timeRange }
);

/** Utility provider for `KBQ_TIME_RANGE_LOCALE_CONFIGURATION`. */
export const kbqTimeRangeLocaleConfigurationProvider = (configuration: KbqTimeRangeLocaleConfig): Provider => ({
    provide: KBQ_TIME_RANGE_LOCALE_CONFIGURATION,
    useValue: configuration
});

@Component({
    standalone: true,
    selector: 'kbq-time-range',
    template: `
        @let localeConfig = localeConfiguration();
        <kbq-time-range-title
            #popover="kbqPopover"
            class="kbq-time-range__trigger"
            kbqPopover
            kbqPopoverClass="kbq-time-range__popover"
            [kbqPopoverSize]="popoverSize"
            [kbqPopoverContent]="timeRangePopoverContent"
            [kbqPopoverFooter]="timeRangePopoverFooter"
            [kbqPopoverPlacement]="popupPlacement"
            [kbqPopoverArrow]="arrow()"
            [titleTemplate]="titleTemplate()"
            [timeRange]="titleValue()"
            [localeConfiguration]="localeConfig"
            (kbqPopoverVisibleChange)="onVisibleChange($event)"
        />

        <ng-template #timeRangePopoverContent>
            <kbq-time-range-editor
                [maxDate]="maxDate()"
                [minDate]="minDate()"
                [formControl]="rangeEditorControl"
                [localeConfiguration]="localeConfig"
                [showRangeAsDefault]="showRangeAsDefault()"
                [rangeValue]="normalizedDefaultRangeValue()"
                [availableTimeRangeTypes]="availableTimeRangeTypes()"
            />
        </ng-template>

        <ng-template #timeRangePopoverFooter>
            <div class="kbq-time-range__buttons" role="group">
                <button
                    kbq-button
                    [color]="'contrast'"
                    [disabled]="rangeEditorControl.invalid"
                    (click)="onApply(popover)"
                >
                    {{ localeConfig.editor.apply }}
                </button>

                <button kbq-button (click)="onCancel(popover)">{{ localeConfig.editor.cancel }}</button>
            </div>
        </ng-template>
    `,
    styleUrls: ['./time-range.scss'],
    imports: [
        ReactiveFormsModule,
        KbqPopoverModule,
        KbqButtonModule,
        KbqTimeRangeEditor,
        KbqTimeRangeTitle
    ],
    host: {
        class: 'kbq-time-range'
    },
    providers: [KbqTimeRangeService],
    changeDetection: ChangeDetectionStrategy.OnPush,
    encapsulation: ViewEncapsulation.None
})
export class KbqTimeRange<T> implements ControlValueAccessor, OnInit {
    private readonly timeRangeService = inject<KbqTimeRangeService<T>>(KbqTimeRangeService);
    private readonly localeService = inject(KBQ_LOCALE_SERVICE, { optional: true });
    readonly ngControl = inject(NgControl, { optional: true, self: true });

    /** The minimum selectable date. */
    readonly minDate = input<T>();
    /** The maximum selectable date. */
    readonly maxDate = input<T>();
    /** provided value of selected range */
    readonly defaultRangeValue = input<KbqRangeValue<T>>();
    /** Preset of selectable ranges */
    readonly availableTimeRangeTypes = input<KbqTimeRangeType[]>(this.timeRangeService.providedDefaultTimeRangeTypes);
    /** Customizable trigger output */
    readonly titleTemplate = input<TemplateRef<KbqTimeRangeCustomizableTitleContext>>();
    /** Whether to show popover with arrow */
    readonly arrow = input(true, { transform: booleanAttribute });
    /**
     * Whether to show range in popover if not provided
     * @see availableTimeRangeTypes
     */
    readonly showRangeAsDefault = input(true);
    /** Whether component should fallback to default value if null provided */
    readonly nonNullable = input(true, { transform: booleanAttribute });

    /** Emit value update if provided value via formControl wasn't valid */
    readonly valueCorrected = output<KbqTimeRangeRange>();

    /**
     * Used to calculate time range.
     * @docs-private */
    protected readonly normalizedDefaultRangeValue = computed(() => ({
        ...this.timeRangeService.getDefaultRangeValue(),
        ...this.defaultRangeValue()
    }));

    /** @docs-private */
    protected titleValue: WritableSignal<KbqTimeRangeRange | null>;
    /** @docs-private */
    protected readonly rangeEditorControl: FormControl<KbqTimeRangeRange>;

    /** @docs-private */
    protected readonly popoverSize = PopUpSizes.Medium;
    /** @docs-private */
    protected readonly popupPlacement = PopUpPlacements.BottomLeft;

    /** @docs-private */
    protected readonly localeConfiguration = signal(inject(KBQ_TIME_RANGE_LOCALE_CONFIGURATION));

    constructor() {
        if (this.ngControl) {
            this.ngControl.valueAccessor = this;
        }

        const defaultValue = this.timeRangeService.getTimeRangeDefaultValue(
            this.normalizedDefaultRangeValue(),
            this.availableTimeRangeTypes()
        );

        this.titleValue = signal(this.nonNullable() ? defaultValue : null);
        this.rangeEditorControl = new FormControl<KbqTimeRangeRange>(defaultValue, { nonNullable: true });

        this.localeService?.changes.pipe(takeUntilDestroyed()).subscribe(() => {
            this.localeConfiguration.set(this.localeService?.getParams('timeRange') ?? ruRULocaleData.timeRange);
        });

        toObservable(this.availableTimeRangeTypes)
            .pipe(takeUntilDestroyed())
            .subscribe(this.handleAvailableTypesChange);
    }

    ngOnInit(): void {
        // call again init, so input signals values will be correct
        this.writeValue(this.ngControl?.value ?? null);
    }

    /** Implemented as part of ControlValueAccessor */
    writeValue(value: KbqTimeRangeRange | null): void {
        const nonNullable = this.nonNullable();

        const availableTimeRangeTypes = this.availableTimeRangeTypes();

        const corrected = this.timeRangeService.checkAndCorrectTimeRangeValue(
            value,
            availableTimeRangeTypes,
            this.normalizedDefaultRangeValue()
        );

        if (
            (value && 'type' in value && !('startDateTime' in value)) ||
            (value && 'type' in value && !availableTimeRangeTypes.includes(value.type)) ||
            (nonNullable && value === null)
        ) {
            this.valueCorrected.emit(corrected);
        }

        this.titleValue.set(nonNullable ? corrected : value === null ? null : corrected);
        this.rangeEditorControl.setValue(corrected);
    }

    /** @docs-private */
    onApply(popover: KbqPopoverTrigger): void {
        this.titleValue.set(this.rangeEditorControl.value);
        this.onChange(this.rangeEditorControl.value);
        popover.hide();
    }

    /** @docs-private */
    onCancel(popover: KbqPopoverTrigger): void {
        popover.hide();
    }

    onVisibleChange(isVisible: boolean) {
        const titleValue = this.titleValue();

        if (!isVisible && titleValue) {
            this.rangeEditorControl.setValue(titleValue);
        }
    }

    /** Implemented as part of ControlValueAccessor
     * @docs-private */
    onChange = (_value: KbqTimeRangeRange) => {};
    /** Implemented as part of ControlValueAccessor
     * @docs-private */
    onTouch = () => {};
    /** Implemented as part of ControlValueAccessor */
    registerOnChange(fn: (_value: KbqTimeRangeRange) => void): void {
        this.onChange = fn;
    }
    /** Implemented as part of ControlValueAccessor */
    registerOnTouched(fn: () => void): void {
        this.onTouch = fn;
    }

    private handleAvailableTypesChange = (types: KbqTimeRangeType[]) => {
        if (types.includes(this.rangeEditorControl.value.type) || this.rangeEditorControl.value.type === 'range') {
            return;
        }

        const timeRangeDefaultValue = this.timeRangeService.getTimeRangeDefaultValue(
            this.normalizedDefaultRangeValue(),
            types.length ? types : ['range']
        );

        this.titleValue.set(timeRangeDefaultValue);
        this.rangeEditorControl.setValue(timeRangeDefaultValue);
        this.onChange(timeRangeDefaultValue);
    };
}
