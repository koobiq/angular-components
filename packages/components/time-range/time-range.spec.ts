import { ChangeDetectionStrategy, Component, DebugElement, Provider, signal, Type } from '@angular/core';
import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { FormControl, ReactiveFormsModule } from '@angular/forms';
import { By } from '@angular/platform-browser';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { KbqLuxonDateModule } from '@koobiq/angular-luxon-adapter/adapter';
import { KbqFormattersModule } from '@koobiq/components/core';
import { KbqPopoverComponent } from '@koobiq/components/popover';
import { KbqRadioButton } from '@koobiq/components/radio';
import { KbqTimeRange } from './time-range';
import { KbqTimeRangeTitle } from './time-range-title';
import { KbqTimeRangeRange, KbqTimeRangeType } from './types';

const setup = <T>(component: Type<T>, providers: Provider[] = []): ComponentFixture<T> => {
    TestBed.configureTestingModule({
        imports: [component, NoopAnimationsModule, KbqLuxonDateModule, KbqFormattersModule],
        providers: [...providers]
    });
    const fixture = TestBed.createComponent<T>(component);

    fixture.autoDetectChanges();

    return fixture;
};

const getTriggerNativeElement = (debugElement: DebugElement): HTMLElement => {
    return debugElement.query(By.directive(KbqTimeRangeTitle)).nativeElement;
};

const getPopoverDebugElement = (debugElement: DebugElement): DebugElement => {
    return debugElement.query(By.directive(KbqPopoverComponent));
};

describe('KbqTimeRange', () => {
    describe('Component initialization', () => {
        it('should create component without errors', () => {
            const { debugElement } = setup(TestComponent);

            expect(debugElement.query(By.directive(KbqTimeRange))).toBeTruthy();
        });

        it('should apply default configuration', () => {
            const { debugElement } = setup(TestComponent);

            expect(getTriggerNativeElement(debugElement).textContent).toMatchSnapshot();
        });

        it('should open popover when trigger is clicked', fakeAsync(() => {
            const fixture = setup(TestComponent);
            const { debugElement } = fixture;
            const triggerElement = getTriggerNativeElement(debugElement);

            triggerElement.click();
            tick();
            fixture.detectChanges();

            expect(getPopoverDebugElement(debugElement)).toBeTruthy();
        }));

        it('should select first radio if no external value provided', fakeAsync(() => {
            const fixture = setup(TestComponent);
            const { debugElement } = fixture;
            const triggerElement = getTriggerNativeElement(debugElement);

            triggerElement.click();
            tick();
            fixture.detectChanges();

            const popoverElement = getPopoverDebugElement(debugElement);

            expect(popoverElement.queryAll(By.directive(KbqRadioButton))[0].classes['kbq-selected']).toBeTruthy();
        }));

        it('should select first radio if availableTimeRangeTypes provided', fakeAsync(() => {
            const fixture = setup(TestComponentWithInputs);
            const { debugElement } = fixture;
            const triggerElement = getTriggerNativeElement(debugElement);

            triggerElement.click();
            tick();
            fixture.detectChanges();

            const popoverElement = getPopoverDebugElement(debugElement);

            expect(
                popoverElement
                    .queryAll(By.directive(KbqRadioButton))
                    .findIndex((element) => element.classes['kbq-selected'])
            ).toBe(0);
        }));

        it('should check selected radio if external value provided', fakeAsync(() => {
            const fixture = setup(TestComponentWithInitial);
            const { debugElement } = fixture;
            const triggerElement = getTriggerNativeElement(debugElement);

            triggerElement.click();
            tick();
            fixture.detectChanges();

            const popoverElement = getPopoverDebugElement(debugElement);
            const checkedRadio: HTMLElement | undefined = popoverElement
                .queryAll(By.directive(KbqRadioButton))
                .find((element) => element.classes['kbq-selected'])?.nativeElement satisfies HTMLElement;

            expect({
                trigger: getTriggerNativeElement(debugElement).textContent,
                checkedRadio: checkedRadio?.textContent
            }).toMatchSnapshot();
        }));

        it('should check range as default if nothing provided', () => {
            const fixture = setup(TestComponentWithInputs);
            const { componentInstance } = fixture;
            const initial = componentInstance.control.value.type;

            componentInstance.availableTimeRangeTypes.set([]);
            fixture.detectChanges();

            expect({
                initial,
                current: componentInstance.control.value.type
            }).toMatchSnapshot();
        });
    });
});

@Component({
    standalone: true,
    template: `
        <kbq-time-range />
    `,
    imports: [KbqTimeRange],
    changeDetection: ChangeDetectionStrategy.OnPush
})
export class TestComponent {}

@Component({
    standalone: true,
    template: `
        <kbq-time-range [formControl]="control" />
    `,
    imports: [KbqTimeRange, ReactiveFormsModule],
    changeDetection: ChangeDetectionStrategy.OnPush
})
export class TestComponentWithInitial {
    control = new FormControl<KbqTimeRangeRange>({ type: 'currentYear' }, { nonNullable: true });
}

@Component({
    standalone: true,
    template: `
        <kbq-time-range [availableTimeRangeTypes]="availableTimeRangeTypes()" [formControl]="control" />
    `,
    imports: [KbqTimeRange, ReactiveFormsModule],
    changeDetection: ChangeDetectionStrategy.OnPush
})
export class TestComponentWithInputs {
    availableTimeRangeTypes = signal<KbqTimeRangeType[]>([
        'lastMinute',
        'last5Minutes',
        'last15Minutes',
        'last30Minutes',
        'lastHour',
        'last24Hours',
        'last3Days',
        'last7Days',
        'last14Days',
        'last30Days',
        'last3Months',
        'last12Months',
        'allTime',
        'currentQuarter',
        'currentYear',
        'range'
    ]);
    control = new FormControl<KbqTimeRangeRange>({ type: this.availableTimeRangeTypes()[0] }, { nonNullable: true });
}
