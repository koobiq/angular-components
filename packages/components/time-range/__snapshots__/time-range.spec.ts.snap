// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KbqTimeRange Component initialization should apply default configuration 1`] = `"за последний час"`;

exports[`KbqTimeRange Component initialization should check range as default if nothing provided 1`] = `
{
  "current": "range",
  "initial": "lastMinute",
}
`;

exports[`KbqTimeRange Component initialization should check selected radio if external value provided 1`] = `
{
  "checkedRadio": " за текущий год ",
  "trigger": "за текущий год",
}
`;
